kernel_gateway/gatewayapp.py

```diff
@@ -494,14 +506,13 @@ def _build_ssl_options(self):
             ssl_options['keyfile'] = self.keyfile
         if self.client_ca:
             ssl_options['ca_certs'] = self.client_ca
+        if self.certfile and self.ssl_version:
```

Comment:
I think we should try to keep ssl_version independent from the related items for determining whether it should be set.

Reply:
But `certfile` key is required on the ssl_options according to the comment I found in the `__init__` function in tornado's `tcpserver.py`. The comment is: `# Only certfile is required: it can contain both keys`. That means, we can't really set `ssl_version` attribute just by itself. If I do that I would get the same `KeyError('missing key "certfile" in ssl_options')`. Right?

Reply:
Thanks for the reference.  While that comment is true, dependent applications shouldn't have to understand the inter-relationships.  You'll get that `KeyError` today if you only set `keyfile` or `ca_certs` as well, and we don't enforce that relationship here or in the tests.

Reply:
Got it. I will remove that dependency.